utils/hare-status

```diff
@@ -167,31 +148,91 @@ def processes(cns: Consul, consul_util: ConsulUtil,
         parts = x['Key'].split('/', 5)
         if len(parts) == 5:
             proc_fidks.append(fid_key(parts[-1]))
+    node_health = consul_util.get_node_health_status(node_name)
+    process_states = get_node_process_states(cns, node_name)
+    process_list = get_node_processes(cns, node_name)
     return [
         Process(
-            name=proc_id2name(cns, node_name, k),
+            name=proc_id2name(process_list, k),
             fid=Fid(0x7200000000000001, k),
-            ep=(
-                kv_value_as_str(
-                    cns,
-                    f'm0conf/nodes/{node_name}/processes/{k}/endpoint'
-                ) or '**ERROR**'),
-            status=process_status(consul_util, node_name,
+            ep=(proc_id2endpoint(process_list, k) or '**ERROR**'),
+            status=process_status(node_health, process_states,
                                   Fid(0x7200000000000001, k)))
         for k in sorted(proc_fidks)
     ]
 
 
-def process_status(consul_util: ConsulUtil, node_name, fid: Fid) -> str:
-    if (consul_util.get_node_health_status(node_name) != 'passing'):
+def process_status(node_health: str, proc_states: Dict[Fid, str],
+                   fid: Fid) -> str:
+    if node_health != 'passing':
         return 'offline'
-    status = consul_util.get_process_status(fid, node_name)
-    if status.proc_status == 'M0_CONF_HA_PROCESS_STARTED':
-        return 'started'
-    elif status.proc_status in ('Unknown', 'UNKNOWN'):
-        return 'unknown'
+    if fid in proc_states:
+        return proc_states[fid]
     else:
-        return 'offline'
+        return 'unknown'
+
+
+def get_node_process_states(cns: Consul, node_name: str) -> Dict[Fid, str]:
+    key = f'{node_name}/processes/'
+    proc_list = kv_item(cns, key, recurse=True)
+    process_states: Dict[Fid, str] = {}
+    for proc in proc_list:
+        key_split = proc['Key'].split('/')
+        if len(key_split) != 3:
+            continue
+        fid = key_split[-1]
+        val = json.loads(proc['Value'])
+        if val['state'] == 'M0_CONF_HA_PROCESS_STARTED':
+            state = 'started'
+        elif val['state'] in ('Unknown', 'UNKNOWN'):
+            state = 'unknown'
+        else:
+            state = 'offline'
+        process_states[Fid.parse(fid)] = state
+    return process_states
+
+
+def proc_id2endpoint(process_list: Any, proc_id: int) -> Optional[str]:
```

Comment:
[optional] I think we should avoid using `Any` when possible.

```suggestion
def proc_id2endpoint(process_list: List[Dict[str, Any]], proc_id: int) -> Optional[str]:
```

Reply:
done